app/services/sendgrid_api.rb

```diff
@@ -105,21 +105,21 @@ def email_found?(response)
   end
 
   def call_api(method, action, data)
-    unless enabled?
-      Rails.logger.error('Sendgrid is disabled')
-      return false
-    end
+    return false unless enabled?
 
     response = @pool.with { |client| client.request(method, action, data) }
 
     yield response
   rescue => e
     Rails.logger.error("#{e.class.name}: #{e.message}")
-    Raven.capture_exception(e) unless RESCUABLE_ERRORS.include?(e.class)
+    unless RESCUABLE_ERRORS.any? { |error_klass| e.is_a?(error_klass) }
+      Raven.capture_exception(e)
+    end
     false
   end
 
   def enabled?
+    Rails.logger.error('Sendgrid is disabled') unless @enabled
```

Comment:
Also, not sure why you moved this out of the call api method? It makes more sense there doesn't it? You calling the API and want to warn that sendgrid is disabled?